app/models/concerns/user_encrypted_attribute_overrides.rb

```diff
@@ -15,7 +15,8 @@ def find_with_email(email)
 
       email = email.downcase.strip
       email_fingerprint = create_fingerprint(email)
-      find_by(email_fingerprint: email_fingerprint)
+      resource = find_by(email_fingerprint: email_fingerprint)
+      resource if resource&.email == email
```

Comment:
This seems like an odd change. Why is it necessary, and how is this related to phone configurations?

Reply:
It is a result of the changes to the `UpdateUser` service. With this change, we're not decrypting the phone number there (good) but that means we aren't raising an attribute encryption error if there's an issue decrypting attributes. So, a user can sign in without needing any attributes decrypted and a seemingly unrelated 500 pops up further down the line.

Reply:
Do you mean we aren't raising an error when trying to *encrypt* the phone when updating the user? Why would we need to decrypt attributes when setting them?

It seems strange to me that we are looking for a match twice. In what case would there be a match for the email fingerprint but `resource.email` would not equal `email`? This new code is 3.62x slower than the old code, and this is a heavily used method in the app.